### ISSUE TITLE
I cannot have multiple plots in the same window from a function

### ISSUE THREAD

**Issue by noerpel** [xraypy/xraylarch#248]
Hello Matt,

I recently upgraded from Python 2.7 to 3.6.5 and installed the latest version of Larch.  I had some scripts I wrote using the older version that are no longer working properly.  I cannot plot in the same plot window from inside a function.   For example if I call the following function:

def plot_test():
	x = linspace(0,10,101)
	y1 = sin(x)
	y2 = cos(x)
	plot(x,y1,  win=1, new = False, force_draw=True)
	plot(x,y2,  win=1, new = False, force_draw=True)
enddef

It opens up two plot windows.  If I copy and paste the code (without the function declaration) into the Larch GUI, or call the script without the function declaration, it works as expected with the two plots in the same window.  The function call will work if I create a plot window in the Larch GUI before calling the function.  I cannot figure out why this isn't working.   Do you have any ideas? 

Also, I don't know if it's related, but when I call Close_All_Windows() from anywhere, I get the error:

  File "C:\Users\mnoerpel\AppData\Local\Continuum\Anaconda3\lib\site-packages\larch\larchlib.py", line 664, in wrapper
    return fcn(*args, **keywords)
  File "C:\Users\mnoerpel\AppData\Local\Continuum\Anaconda3\share\larch\plugins\wx\plotter.py", line 834, in _closeDisplays
    for name in names:

RuntimeError: dictionary changed size during iteration

and then if I try to plot again, I get the error:

  File "C:\Users\mnoerpel\AppData\Local\Continuum\Anaconda3\lib\site-packages\larch\larchlib.py", line 664, in wrapper
    return fcn(*args, **keywords)
  File "C:\Users\mnoerpel\AppData\Local\Continuum\Anaconda3\share\larch\plugins\wx\plotter.py", line 417, in _plot
    plotter.Raise()

RuntimeError: wrapped C/C++ object of type PlotDisplay has been deleted

Thanks for your help and also making Larch,  It has been a huge help.

Thanks,

Matt Noerpel

**Comment by newville** [comment]
@noerpel OK, that seems really weird to me, but I can reproduce this.  

It goes away (and works as expected) if  I indent the text of "plot_test()", which I don't fully understand either-- that is I don't know why it would be different in the GUI and CLI or what the heck is happening to make a second "window=1".  I sort of expect that the rest of the problems with deleting and recreating plot windows are related to having two windows labeled "Plot Window 1".

Will investigate...

**Comment by noerpel** [comment]
Thanks Matt,

In the original script I wrote the text was indented, I don't know why it wasn't in my question.  I tried changing the indentation in different ways, but it does not help.  I should have mentioned in my original question that I am using a Windows 10 computer issued by the USEPA which has a lot of security software on it.  The anaconda install is in the local directory because I don't have administrative privileges. 

Thanks,

Matt
